source/fundamentals/typescript.txt

```diff
@@ -79,81 +84,119 @@ You can find a code snippet that shows how to specify a type for the ``FindCurso
 
-   .. code-block:: json
+         database
+           .collection<RecursivePet>("your-collection")
+           .findOne({ pet: "Spot" });
 
-      { field: { s1: "hi", s2: "bye" } }
+   .. tab:: Error
+      :tabid: error
 
-   The following query returns no results from this collection, as the value of
-   ``field`` does not exactly match ``{ s1: "hi" }``:
+      .. code-block:: text
+                  
+         index.ts(19,59): error TS2769: No overload matches this call.
+         The last overload gave the following error.
+         Type 'string' is not assignable to type 'Condition<Pet>'.
 
-   .. literalinclude:: /code-snippets/typescript/note-on-dot-notation.ts
-      :language: typescript
-      :linenos:
-      :start-after: start-no-doc
-      :end-before: end-no-doc
+If you must have type safety within nested instances of recursive types,
+you must write your query or update without dot notation.
 
-   The following queries both return your document:
+To learn more about dot notation, see 
+:manual:`Dot Notation </core/document/#dot-notation>`
+in the MongoDB manual.
 
-   .. literalinclude:: /code-snippets/typescript/note-on-dot-notation.ts
-      :language: typescript
-      :linenos:
-      :start-after: start-doc
-      :end-before: end-doc
+.. _node-driver-limitations-mutual-recursion:
 
-   The syntax of the query that does not use dot notation is cumbersome and hard
-   to understand, and may not be worth the type safety obtained from
-   avoiding dot notation.
+Mutual Recursion
+~~~~~~~~~~~~~~~~
+
+You cannot specify a mutually recursive type as a type parameter in version
+{+version+} of the driver.
+
+If you specify a mutually recursive type, the TypeScript compiler raises the
+following error:
+
+.. code-block:: text
+
+   error TS2615: Type of property 'r' circularly references itself in mapped type '{ [Key in keyof MutuallyRecursive]: MutuallyRecursive[Key] extends MutuallyRecursive ? [Key] : MutuallyRecursive extends MutuallyRecursive[Key] ? [...] : MutuallyRecursive[Key] extends readonly (infer ArrayType)[] ? MutuallyRecursive extends ArrayType ? [...] : ArrayType extends MutuallyRecursive ? [...] : [...] : [...'.
+
+A mutually recursive type exists when two types define themselves relative
+to each other. You can update the
+:ref:`Pet <mongodb-node-typescript-pet-interface>` interface
+to be mutually recursive by allowing a pet to have a handler, and defining a
+handler to have a pet. The following is the mutually
+recursive ``Pet`` interface:
+
+.. code-block:: typescript
+   :emphasize-lines: 2, 8
+
+   interface MutuallyRecursivePet {
+      handler?: Handler;
+      name: string;
+      age: number;
+   }
+
+   interface Handler {
+      pet: MutuallyRecursivePet;
+      name: string;
+   }
 
-For more information on dot notation, see :manual:`the MongoDB Manual </core/document/#dot-notation>`.
+If you must apply a mutually recursive type to your classes, use version 4.2 of
+the {+driver-long+}.
```

Comment:
If I recall correctly, I think we decided as a team to only use the long version of the name if it both appears in an intro section and is the first mention of it.

Reply:
Similar comment for other instances.